Source/Phone/Call/CallManager.swift

```diff
@@ -89,35 +100,42 @@ class CallManager {
         
         // If it belongs to existing active call, update it.
         if let call = callInstances[callUrl] {
-			call.update(callInfo: callInfo)
-            return
-        }
-        
-        if callInfo.isIncomingCall {
-            doActionWhenIncoming(callInfo)
-        } else if callInfo.hasJoinedOnOtherDevice {
-            doActionWhenJoinedOnOtherDevice(callInfo)
+            call.update(callInfo: callInfo)
+        } else if let deviceUrlString = deviceService.deviceUrl,
+            let deviceUrl = URL(string: deviceUrlString),
+            callInfo.isIncomingCall,
+            callInfo.hasJoinedOnOtherDevice(deviceUrl: deviceUrl) {
+            // XXX: Is this conditional intended to add this call even when there is no real device registration information?
+            // At the time of writing the deviceService.deviceUrl will return a saved value from the UserDefaults. When the application
+            // has been restarted and the reregistration process has not completed, other critical information such as locusServiceUrl
+            // will not be available, but the deviceUrl WILL be. This may put the application in a bad state. This code MAY be dealing with
+            // a race condition and this MAY be the solution to not dropping a call before reregistration has been completed. 
+            // If so it needs improvement, if not it may be able to be dropped.
+            let callClient = CallClient(authenticationStrategy: authenticationStrategy, deviceService: deviceService)
+            let call = Call(callInfo, callManager: self, callClient: callClient, deviceUrl: deviceUrl, callMetrics: callMetrics)
+            addCallWith(url: callUrl, call: call)
+
+            if callInfo.isIncomingCall {
+                callNotificationCenter.notifyIncomingCall(call)
+                Logger.info("Receive incoming call")
+
+                // Intentional use of deprecated API for backwards compatibility
+                PhoneNotificationCenter.sharedInstance.notifyIncomingCall(call)
+            }
+            // TODO: need to support other device joined case
         }
     }
     
-    private func handleActiveCalls(_ callInfos: [CallInfo]) {
-        for callInfo in callInfos {
-			handle(callInfo: callInfo)
-        }
+    func createOutgoingCall() -> Call {
+        let callClient = CallClient(authenticationStrategy: authenticationStrategy, deviceService: deviceService)
+        return Call(callManager: self, callClient: callClient, deviceUrl: deviceService.device!.deviceUrl, callMetrics: callMetrics)
```

Comment:
Yes sir! It did before that as well. We've not eliminated all instances of that, hopefully we've knocked out a lot of them and made many others more visible though.